### ISSUE TITLE
lxc-attach  fail to start the process : Invalid argument - Failed to open terminal  or  Inappropriate ioctl for device

### ISSUE THREAD

**Issue by john1jxm** [lxc/lxc#3447]
1  lxc-3.0.0
container config:
lxc.signal.halt = SIGUSR1
lxc.signal.reboot = SIGTERM
lxc.uts.name = NOS
lxc.console.path = none
lxc.init.cmd = /sbin/ContainerInit
lxc.mount.auto = cgroup:mixed proc:mixed
lxc.pty.max = 8
lxc.tty.max = 1
lxc.cgroup.memory.limit_in_bytes = 80M
lxc.cgroup.cpu.cfs_period_us = 100000
lxc.cgroup.cpu.cfs_quota_us = 60000
lxc.namespace.keep = ipc net user cgroup
lxc.environment = APP_ROOT=/flash/opkg

2   lxc-start failure log:
```
lxc-start NOS 19700101000317.429 INFO     lxc_lsm - lsm/lsm.c:lsm_init:46 - LSM security driver nop
lxc-start NOS 19700101000317.430 INFO     terminal - terminal.c:lxc_terminal_setup:911 - No terminal requested
lxc-start NOS 19700101000317.430 INFO     lxc_start - start.c:lxc_init:846 - Container "NOS" is initialized
lxc-start NOS 19700101000317.430 ERROR    lxc_start - start.c:lxc_spawn:1553 - Failed initializing cgroup support
lxc-start NOS 19700101000317.431 ERROR    lxc_start - start.c:__lxc_start:1866 - Failed to spawn container "NOS"
lxc-start NOS 19700101000348.713 INFO     lxc_lsm - lsm/lsm.c:lsm_init:46 - LSM security driver nop
lxc-start NOS 19700101000348.714 INFO     terminal - terminal.c:lxc_terminal_setup:911 - No terminal requested
lxc-start NOS 19700101000348.714 INFO     lxc_start - start.c:lxc_init:846 - Container "NOS" is initialized
lxc-start NOS 19700101000348.714 ERROR    lxc_start - start.c:lxc_spawn:1553 - Failed initializing cgroup support
lxc-start NOS 19700101000348.714 ERROR    lxc_start - start.c:__lxc_start:1866 - Failed to spawn container "NOS"
lxc-start NOS 19700101000441.264 INFO     lxc_lsm - lsm/lsm.c:lsm_init:46 - LSM security driver nop
lxc-start NOS 19700101000441.298 INFO     terminal - terminal.c:lxc_terminal_setup:911 - No terminal requested
lxc-start NOS 19700101000441.306 INFO     lxc_start - start.c:lxc_init:846 - Container "NOS" is initialized
lxc-start NOS 19700101000441.322 ERROR    lxc_start - start.c:lxc_spawn:1553 - Failed initializing cgroup support
lxc-start NOS 19700101000441.356 ERROR    lxc_start - start.c:__lxc_start:1866 - Failed to spawn container "NOS"
lxc-start NOS 19700101000459.641 INFO     lxc_lsm - lsm/lsm.c:lsm_init:46 - LSM security driver nop
lxc-start NOS 19700101000459.642 INFO     terminal - terminal.c:lxc_terminal_setup:911 - No terminal requested
lxc-start NOS 19700101000459.642 INFO     lxc_start - start.c:lxc_init:846 - Container "NOS" is initialized
lxc-start NOS 19700101000459.642 INFO     lxc_cgroup - cgroups/cgroup.c:cgroup_init:60 - cgroup driver cgfsng initing for NOS
lxc-start NOS 19700101000459.643 DEBUG    lxc_cgfsng - cgroups/cgfsng.c:cg_legacy_filter_and_set_cpus:542 - The path "/sys/devices/system/cpu/isolated" to read isolated cpus from does not exist
lxc-start NOS 19700101000459.643 DEBUG    lxc_cgfsng - cgroups/cgfsng.c:cg_legacy_filter_and_set_cpus:549 - Copying cpu settings of parent cgroup
lxc-start NOS 19700101000459.648 INFO     lxc_start - start.c:lxc_spawn:1614 - Cloned CLONE_NEWNS
lxc-start NOS 19700101000459.648 INFO     lxc_start - start.c:lxc_spawn:1614 - Cloned CLONE_NEWPID
lxc-start NOS 19700101000459.648 INFO     lxc_start - start.c:lxc_spawn:1614 - Cloned CLONE_NEWUTS
lxc-start NOS 19700101000459.649 DEBUG    lxc_start - start.c:lxc_try_preserve_namespaces:199 - Preserved mnt namespace via fd 10
lxc-start NOS 19700101000459.649 DEBUG    lxc_start - start.c:lxc_try_preserve_namespaces:199 - Preserved pid namespace via fd 11
lxc-start NOS 19700101000459.649 DEBUG    lxc_start - start.c:lxc_try_preserve_namespaces:199 - Preserved uts namespace via fd 12
lxc-start NOS 19700101000459.651 DEBUG    lxc_cgfsng - cgroups/cgfsng.c:__cg_legacy_setup_limits:2735 - Set controller "memory.limit_in_bytes" set to "80M"
lxc-start NOS 19700101000459.651 DEBUG    lxc_cgfsng - cgroups/cgfsng.c:__cg_legacy_setup_limits:2735 - Set controller "cpu.cfs_period_us" set to "100000"
lxc-start NOS 19700101000459.651 DEBUG    lxc_cgfsng - cgroups/cgfsng.c:__cg_legacy_setup_limits:2735 - Set controller "cpu.cfs_quota_us" set to "60000"
lxc-start NOS 19700101000459.651 INFO     lxc_cgfsng - cgroups/cgfsng.c:__cg_legacy_setup_limits:2740 - Limits for the legacy cgroup hierarchies have been setup
lxc-start NOS 19700101000459.653 DEBUG    lxc_start - start.c:lxc_spawn:1668 - Preserved net namespace via fd 6
lxc-start NOS 19700101000459.654 INFO     lxc_conf - conf.c:setup_utsname:774 - Set hostname to "NOS"
lxc-start NOS 19700101000459.654 INFO     lxc_conf - conf.c:mount_autodev:1163 - Preparing "/dev"
lxc-start NOS 19700101000459.655 ERROR    lxc_utils - utils.c:open_without_symlink:1598 - Too many levels of symbolic links - dev in /dev was a symbolic link!
lxc-start NOS 19700101000459.655 ERROR    lxc_conf - conf.c:mount_autodev:1182 - Too many levels of symbolic links - Failed to mount tmpfs on "/dev"
lxc-start NOS 19700101000459.655 ERROR    lxc_conf - conf.c:lxc_setup:3338 - Failed to mount "/dev"
lxc-start NOS 19700101000459.656 ERROR    lxc_start - start.c:do_start:1198 - Failed to setup container "NOS"
lxc-start NOS 19700101000459.657 ERROR    lxc_sync - sync.c:__sync_wait:57 - An error occurred in another process (expected sequence number 5)
lxc-start NOS 19700101000459.658 ERROR    lxc_start - start.c:__lxc_start:1866 - Failed to spawn container "NOS"

3   add lxc.autodev = 0 into config
 success to lxc-start the container 
but lxc-attach  fail

4   lxc-attach failure log:
lxc-attach NOS 19700101001945.839 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_init_pid" is 0
lxc-attach NOS 19700101001945.839 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_init_pid" is 0
lxc-attach NOS 19700101001945.841 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_config_item" is 0
lxc-attach NOS 19700101001945.842 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_config_item" is 0
lxc-attach NOS 19700101001945.842 INFO     lxc_attach - attach.c:fetch_seccomp:679 - Failed to get running config item for lxc.seccomp.profile
lxc-attach NOS 19700101001945.842 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_config_item" is 0
lxc-attach NOS 19700101001945.842 INFO     lxc_attach - attach.c:fetch_seccomp:683 - Failed to get running config item for lxc.seccomp
lxc-attach NOS 19700101001945.843 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_clone_flags" is 0
lxc-attach NOS 19700101001945.843 DEBUG    lxc_attach - attach.c:lxc_attach:1170 - Inheriting user namespace from 6224
lxc-attach NOS 19700101001945.844 DEBUG    lxc_attach - attach.c:lxc_attach:1170 - Inheriting ipc namespace from 6224
lxc-attach NOS 19700101001945.844 DEBUG    lxc_attach - attach.c:lxc_attach:1170 - Inheriting net namespace from 6224
lxc-attach NOS 19700101001945.844 ERROR    terminal - terminal.c:lxc_terminal_create:876 - Inappropriate ioctl for device - Failed to open terminal
lxc-attach NOS 19700101001945.844 ERROR    lxc_attach - attach.c:lxc_attach_terminal:986 - Inappropriate ioctl for device - Failed to create terminal
lxc-attach NOS 19700101001945.844 ERROR    lxc_attach - attach.c:lxc_attach:1194 - Failed to setup new terminal
```

**Comment by john1jxm** [comment]
**Invalid argument error**

lxc-attach NOS 19700101005227.496 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_init_pid" is 0
lxc-attach NOS 19700101005227.496 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_init_pid" is 0
lxc-attach NOS 19700101005227.496 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_config_item" is 0
lxc-attach NOS 19700101005227.497 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_config_item" is 0
lxc-attach NOS 19700101005227.497 INFO     lxc_attach - attach.c:fetch_seccomp:679 - Failed to get running config item for lxc.seccomp.profile
lxc-attach NOS 19700101005227.497 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_config_item" is 0
lxc-attach NOS 19700101005227.497 INFO     lxc_attach - attach.c:fetch_seccomp:683 - Failed to get running config item for lxc.seccomp
lxc-attach NOS 19700101005227.497 DEBUG    lxc_commands - commands.c:lxc_cmd_rsp_recv:160 - Response data length for command "get_clone_flags" is 0
lxc-attach NOS 19700101005227.497 DEBUG    lxc_attach - attach.c:lxc_attach:1170 - Inheriting user namespace from 27777
lxc-attach NOS 19700101005227.497 DEBUG    lxc_attach - attach.c:lxc_attach:1170 - Inheriting ipc namespace from 27777
lxc-attach NOS 19700101005227.498 DEBUG    lxc_attach - attach.c:lxc_attach:1170 - Inheriting net namespace from 27777
lxc-attach NOS 19700101005227.498 ERROR    terminal - terminal.c:lxc_terminal_create:876 - Invalid argument - Failed to open terminal
lxc-attach NOS 19700101005227.498 ERROR    lxc_attach - attach.c:lxc_attach_terminal:986 - Invalid argument - Failed to create terminal
lxc-attach NOS 19700101005227.498 ERROR    lxc_attach - attach.c:lxc_attach:1194 - Failed to setup new terminal
[root@AONT: /]#

**Comment by brauner** [comment]
This issue is a bit jumbled so it's hard to follow what the exact problem is and crucial information about what host, kernel, and lxc version you're exactly using is missing. Could you please fill in this info? Please make sure to reproduce with at least 3.0.4.

**Comment by brauner** [comment]
Haven't heard back in a while. If you have more details, feel free to open a new issue.
